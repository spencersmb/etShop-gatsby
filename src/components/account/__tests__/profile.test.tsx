import React from 'react'
import renderer from 'react-test-renderer'
import {
	render,
	cleanup
} from 'react-testing-library'
import 'jest-dom/extend-expect'
import Profile from '../profile'

afterEach(cleanup)

const hasUser = {
	user: {
		email: 'spencer.bigum@gmail.com',
		firstName: 'Spencer',
		lastName: 'spencer',
		token: '12345'
	}
}
describe('Profile Layout', () => {

	it('renders correctly', () => {
		const tree = renderer
			.create(
				<Profile {...hasUser}/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('Should render correct user email', () => {
		const modalRender = render(<Profile {...hasUser}/>)
		expect(modalRender.getByText('spencer.bigum@gmail.com').innerHTML).toEqual('spencer.bigum@gmail.com')
	})

})