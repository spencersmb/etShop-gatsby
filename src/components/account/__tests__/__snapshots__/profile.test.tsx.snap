// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Layout renders correctly 1`] = `
Array [
  <h1>
    Your profile
  </h1>,
  <ul>
    <li>
      Name: 
      Spencer
    </li>
    <li>
      E-mail: 
      <span>
        spencer.bigum@gmail.com
      </span>
    </li>
  </ul>,
]
`;
